File: Лабораторная работа 1/№2-метод Ньютона.cpp
#include <iostream>
#include <cmath>

double func(double x) {
    return exp(-x) - sqrt(x - 1);
}

double deriv(double x) {
    return -exp(-x) - 1 / (2 * sqrt(x - 1));
}

double newton(double x0, double eps) {
    double x = x0;
    double next_x;
    int iter = 0;

    do {
        next_x = x - func(x) / deriv(x);
        iter++;
        x = next_x;
    } while (std::abs(func(next_x)) >= eps);

    return next_x;
}

int main() {
    double x0 = 2.0;
    double eps = 1e-6;

    double root = newton(x0, eps);

    std::cout << "Root: " << root << std::endl;

    return 0;
}

File: Лабораторная работа 1/№2-метод Ньютона.py
def f_prime(x):
    return -math.exp(-x) - 1 / (2 * math.sqrt(x - 1))

def newton_method(x0, eps):
    x1 = x0 - f(x0) / f_prime(x0)
    
    while abs(x1 - x0) > eps:
        x0 = x1
        x1 = x0 - f(x0) / f_prime(x0)

    return x1

x0 = 1.5
eps = 1e-6

root = newton_method(x0, eps)
print(f"Root found at x = {root}")
